packages/txservice/src/provider.ts

```diff
@@ -0,0 +1,177 @@
+    this.config = { gasInitialBumpPercent, gasPriceMinimum, rpcProviderMaxRetries };
+    this.confirmationsRequired = config.chainConfirmations.get(chainId) ?? config.defaultConfirmationsRequired;
+    this.chainId = chainId;
+    
+    // Register a provider for each url.
+    // Make sure all providers are ready()
+
+    if (providerConfigs.length === 1) {
+      this.provider = new JsonRpcProvider(providerConfigs[0]);
+    } else {
+      const fallbackProviderConfigs = providerConfigs.map((config) => ({
+        provider: new JsonRpcProvider(
+          {
+            url: config.url,
+            user: config.user,
+            password: config.password,
+          }
+        ),
+        priority: config.priority ?? 1,
+        weight: config.weight ?? 1,
+        stallTimeout: config.stallTimeout,
+      }));
+      // TODO: Quorum is set to 1 here, but we may want to reconfigure later. Normally it is half the sum of the weights,
+      // which might be okay in our case, but for now we have a low bar.
+      this.provider = new FallbackProvider(fallbackProviderConfigs, 1);
+    }
+
+    this.signer = typeof signer === "string" ? new Wallet(signer, this.provider) : signer.connect(this.provider);
+  }
+
+  public async sendTransaction(
+    tx: FullTransaction
+  ): Promise<{ response: providers.TransactionResponse | Error; success: boolean }> {
+    const ready = await this.provider.ready;
+    if (!ready) {
+      // Error out, no providers are ready.
+      throw new ChainError(ChainError.reasons.ProviderNotReady);
+    }
+    // TODO: Check getBlockNumber() > 0.
+
+    // Define task to send tx with proper nonce
+    const task = async (): Promise<{ response: providers.TransactionResponse | Error; success: boolean }> => {
+      try {
+        // Send transaction using the passed in callback.
+        // const stored = this.nonces.get(chainId);
+        // const nonceToUse: number = nonce ?? stored ?? (await signer.getTransactionCount("pending"));
+        const pending: number = await this.signer.getTransactionCount("pending");
+        const { to, data, chainId, value, gasPrice, nonce } = tx;
+        const response: providers.TransactionResponse | undefined = await this.signer.sendTransaction({
+          to,
+          data,
+          chainId,
+          gasPrice,
+          nonce: nonce ?? pending,
+          value: BigNumber.from(value || 0),
+        });
+        // After calling tx fn, set nonce to the greatest of
+        // stored, pending, or incremented
+        // const pending = await signer.getTransactionCount("pending");
+        // const incremented = (response?.nonce ?? nonceToUse) + 1;
+        // Ensure the nonce you store is *always* the greatest of the values
+        // const toCompare = stored ?? 0;
+        // if (toCompare < pending || toCompare < incremented) {
+        //   this.nonces.set(chainId, incremented > pending ? incremented : pending);
+        // }
```

Comment:
Teehee this was my trashy code, lets update it to the nicer stuff instead, [here](https://github.com/connext/vector/blob/da603eb661bf888a267d802ba315e27c6c0ae028/modules/contracts/src.ts/services/ethService.ts#L210-L232)

:)